addons/mail/models/mail_channel.py

```diff
@@ -145,6 +145,7 @@ def _inverse_channel_partner_ids(self):
         if new_members:
             self.env['mail.channel.partner'].create(new_members)
         if outdated:
+            outdated.refresh()
```

Comment:
This is a deprecated method, but even with invalidate cache the fix does not look correct.

"you are not allowed to delete 'Listeners of a Channel' (mail.channel.partner) records" looks like a missing sudo on the unlink line?

Reply:
The patched code worked well and fixed this issue.
To debug the issue, I added below code to show the value before unlink():

print(outdated.channel_id.channel_partner_ids)

you will see the value of array does NOT contain current login user.
Using refresh() can fix this, you might find another way to do it.

Reply:
If refresh() is deprecated, we can replace it by invalidate_cache()

Reply:
It just so happens that access rules on channel are written with `channel_partner_ids` (the field being updated here) rather than actually using the true source of truth (mail.channel.partner records).

So what your fix is doing is letting the ORM wrongly assume the current user is still a member of the channel when it is already not, or about to not be.

Note that using `invalidate_cache` without using `flush` before would lose any change not yet sent to the database, which is usually incorrect.

But as I said above, the proper fix is to just use `sudo()` on the `unlink()` line instead, to by-pass access rights in this case with the "proper" way, rather than messing with the cache.

Also it would be nice to add a test to cover that method (`action_unfollow`), and also check if maybe the fix shouldn't be done in earlier versions too (not with "start a meeting", but with creating a private channel).

Reply:
Thanks, I'll try to refine it as you suggested and submit a new PR later.